extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/SourceConnectorWrapper.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.kafka.connect.impl;
+
+import com.hazelcast.core.HazelcastException;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.kafka.connect.impl.message.TaskConfigPublisher;
+import com.hazelcast.jet.kafka.connect.impl.message.TaskConfigMessage;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.topic.Message;
+import org.apache.kafka.connect.connector.ConnectorContext;
+import org.apache.kafka.connect.source.SourceConnector;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.source.SourceTask;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static com.hazelcast.client.impl.protocol.util.PropertiesUtil.toMap;
+import static com.hazelcast.internal.util.Preconditions.checkRequiredProperty;
+import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static com.hazelcast.jet.impl.util.ReflectionUtils.newInstance;
+
+/**
+ * This class wraps a Kafka Connector and TaskRunner
+ */
+public class SourceConnectorWrapper {
+    // The logger is accessed by multiple threads
```

Comment:
I don't get it why you need `volatile`; it'd be needed if you would swap loggers, but logger for each wrapper should be final.